### PR TITLE
IDS-2652

### PR DESCRIPTION
**Pull request by illumon-public**
Changes to support customization of ticks shown.

### CODE REVIEW

File: java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java [hunk 1]
```diff
@@ -34,8 +35,25 @@ public void setTransform(BasicAxisTransform transform) {
         this.transform = transform == null ? newNullAxisTransform() : transform;
     }
 
+    public BasicAxisTransform getTransform() {
+        return transform;
+    }
+
+    public Range getDataRange() {
+        if (dataRange == null) {
```
Comment (chipkent) [inline java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java]: Do we always want to return the existing range?  What happens if the axis gets autoranged?  Will we be using the right range?

File: java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java [hunk 2]
```diff
@@ -34,8 +35,25 @@ public void setTransform(BasicAxisTransform transform) {
...
+            if (this.getPlot() instanceof ValueAxisPlot) {
+                final ValueAxisPlot vap = (ValueAxisPlot) this.getPlot();
+                Range r = vap.getDataRange(this);
+                dataRange = r;
```
Comment (chipkent) [inline java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java]: r variable is not necessary.

File: java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java [hunk 3]
```diff
@@ -342,24 +371,25 @@ private boolean contains(Range range, double currentTickValue) {
...
         final Plot plot = this.getPlot();
-        if(plot != null) {
-            if(plot instanceof ValueAxisPlot) {
-                final ValueAxisPlot vap = (ValueAxisPlot)plot;
+        if (plot != null) {
+            if (plot instanceof ValueAxisPlot) {
+                final ValueAxisPlot vap = (ValueAxisPlot) plot;
                 Range r = vap.getDataRange(this);
-                if(r == null) {
+                dataRange = r;
```
Comment (chipkent) [inline java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java]: r variable is not necessary.

File: java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java [hunk 4]
```diff
@@ -324,12 +346,19 @@ private NumberTick createTick(final TickType tickType, final RectangleEdge edge,
...
 
-    private boolean okToPlotTick(final double tickValue){
+    /**
+     * Checks whether the tickValue is valid to draw
+     *
+     * @param tickValue - tick value to check
+     * @param isMajor   - whethe tick is major or minor
+     * @return boolean indicating whether the tick value is valid to show
+     */
+    protected boolean okToPlotTick(final double tickValue, final boolean isMajor) {
```
Comment (devonpa) [inline java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java]: What does this isMajor do? Doesn't seem used

Comment (illumon-public) [reply java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java]: It basically shows whether the tick is major or minor. It is used in iris project. In iris, for DateTimeAxis with BusinessCalendar transformation, if we the tick is major, we're avoiding it and just processing minor ticks.

File: java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java (hunk 1 above)
Comment (illumon-public) [reply java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java]: dataRange is getting updated in autoAdjustRange() function also.

**Pull request merged**